### ISSUE TITLE
When I use manual optimization lightning still check optimizer_idx argument.

### ISSUE THREAD

**Issue by LuoXin-s** [PyTorchLightning/pytorch-lightning#6803]
## 🐛 Bug
I set `self.automatic_optimization = False` in `__init__` like that in [official docs](https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#manual-optimization), but still got an error:

**ValueError: Your LightningModule defines 2 optimizers but training_step is missing the "optimizer_idx" argument.**

This is confusion, since in example there are no `optimizer_idx` in `training_step `, also I think there are no need for it.

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by ananthsub** [comment]
What version of Lightning are you using? In master, this should not raise an exception if you're using manual optimization and if your `training_step` doesn't include `optimizer_idx` https://github.com/PyTorchLightning/pytorch-lightning/blob/bb9ace43334ad50e3758d9cff08ad34216c7d4da/pytorch_lightning/trainer/training_loop.py#L765-L777

**Comment by LuoXin-s** [comment]
@ananthsub I use lightning 1.2.5.

**Comment by akihironitta** [comment]
@LuoXin-s The change #6093 hasn't been released yet. It should be included in 1.3.

**Comment by tchaton** [comment]
Dear @LuoXin-s,

As answered by @akihironitta, the change is only present in 1.3.

Closing the issue. Feel free to re-open it if needed.

Best,
T.C
